packages/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.nuspec

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <metadata>
+        <id>Microsoft.SqlTools.ManagedBatchParser</id>
+        <version>1.0.0</version>
+        <authors>Microsoft</authors>
+        <projectUrl>https://github.com/Microsoft/sqltoolsservice/</projectUrl>
+        <license type="expression">MIT</license>
+        <iconUrl>http://s.gravatar.com/avatar/6f39d8562df0a3509a8240fb281bc5bd?s=80</iconUrl>
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <description>Managed Batch parser for Sql Tools Services.</description>
+        <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+        <tags>SQL Tools XPLAT Managed Batch Parser</tags>
+        <dependencies>
+            <group targetFramework="net472">
+                <dependency id="Microsoft.Data.SqlClient" version="5.0.1" />
+                <dependency id="Microsoft.SqlServer.SqlManagementObjects" version="170.8.0" />
+                <dependency id="Newtonsoft.Json" version="13.0.1" />
+                <dependency id="System.Configuration.ConfigurationManager" version="6.0.0" />
+            </group>
+            <group targetFramework="net6.0">
+                <dependency id="Microsoft.Data.SqlClient" version="5.0.1" />
+                <dependency id="Microsoft.SqlServer.SqlManagementObjects" version="170.8.0" />
+                <dependency id="Newtonsoft.Json" version="13.0.1" />
+                <dependency id="System.Configuration.ConfigurationManager" version="6.0.0" />
+            </group>
+        </dependencies>
+    </metadata>
+    <files>
```

Comment:
> <files> [](http://example.com/codeflow?start=3&length=8)

Is a simple
  <IsPackable>True</IsPackable>
in the .csproj file + a handful properties for the nuget pack target not good enough for you?

This seems like a straight-off-the-book care where you want to delegate all the package management to msbuild and not be in the bisiness of hand-crafting nuspec files

Reply:
+1 to putting everything in csproj, it's much easier to use and modify at build time and lets you share common properties so much easier.

Reply:
Packing is not done on csproj since we need to sign the built binaries first, that's why we need an extra step. 
We would need either another csproj, like there are in packages folder or nuspec. Nuspec is the best way to go for IMO.

Reply:
Here's ours for sql binding - easy to pull these properties out into a common props file for this repo

https://github.com/Azure/azure-functions-sql-extension/blob/main/src/Microsoft.Azure.WebJobs.Extensions.Sql.csproj

Reply:
Build first, then sign, then package. 

Example : 

[Build](https://github.com/Azure/azure-functions-sql-extension/blob/main/builds/azure-pipelines/template-steps-build-test.yml#L67) (GeneratePackageOnBuild=false)

[Package](https://github.com/Azure/azure-functions-sql-extension/blob/main/builds/azure-pipelines/template-steps-build-test.yml#L236) (--no-build)

Reply:
signing of DLLs should happen in one task of the pipeline, then you run
```
msbuild /p:nobuild=1 /t:pack 
or
dotnet pack -nobuild (not sure of the exact syntax/params)
```
so it creates the nuget package with the signed DLLs

then you have to sign the nupkg using esrp so it can go to nuget.org

Reply:
> [Package](https://github.com/Azure/azure-functions-sql-extension/blob/main/builds/azure-pipelines/template-steps-build-test.yml#L236) (--no-build)

We'd still need "signed" binaries in the same location to provide 'no-build'. DLLs are published to artifacts/publish directory, where they are signed and packaged from. 

I don't disagree, but with current setup I don't want to disturb lot of things. As of now separate csproj files in [packages](https://github.com/microsoft/sqltoolsservice/tree/main/packages) folder are used for packaging (view readme), and the only thing I did differently for this NuGet was to use nuspec instead.